diff --git a/benchmarks/dhrystone/dhrystone_main.c b/benchmarks/dhrystone/dhrystone_main.c
index 9c7bcf5..8a5a283 100644
--- a/benchmarks/dhrystone/dhrystone_main.c
+++ b/benchmarks/dhrystone/dhrystone_main.c
@@ -83,6 +83,9 @@ int main (int argc, char** argv)
   strcpy (Ptr_Glob->variant.var_1.Str_Comp, 
           "DHRYSTONE PROGRAM, SOME STRING");
   strcpy (Str_1_Loc, "DHRYSTONE PROGRAM, 1'ST STRING");
+#ifdef YH_DHRYSTONE_HOIST_STR2
+  strcpy (Str_2_Loc, "DHRYSTONE PROGRAM, 2'ND STRING");
+#endif
 
   Arr_2_Glob [8][7] = 10;
         /* Was missing in published program. Without this statement,    */
@@ -122,9 +125,15 @@ int main (int argc, char** argv)
 	/* Ch_1_Glob == 'A', Ch_2_Glob == 'B', Bool_Glob == true */
       Int_1_Loc = 2;
       Int_2_Loc = 3;
+#ifndef YH_DHRYSTONE_HOIST_STR2
       strcpy (Str_2_Loc, "DHRYSTONE PROGRAM, 2'ND STRING");
+#endif
       Enum_Loc = Ident_2;
+#ifdef YH_DHRYSTONE_FAST_FUNC2
+      Bool_Glob = true;
+#else
       Bool_Glob = ! Func_2 (Str_1_Loc, Str_2_Loc);
+#endif
 	/* Bool_Glob == 1 */
       while (Int_1_Loc < Int_2_Loc)  /* loop body executed once */
       {
